src/main/java/codeu/controller/ActivityfeedServlet.java

```diff
@@ -0,0 +1,68 @@
+package codeu.controller;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.UserStore;
+
+/** This class was mostly modeled using the ConversationServlet as a guide */
+public class ActivityfeedServlet extends HttpServlet{
+	
+	/** Store class that gives access to Users. */
+	private UserStore userStore;
+	
+	/** Store class that gives access to Conversations. */
+	private ConversationStore conversationStore;
+	
+	/**
+	 * Set up state for handling the activity feed. This method is only called when
+	 * running in a server, not when running in a test.
+	 */
+	@Override
+	public void init() throws ServletException {
+		super.init();
+	    setUserStore(UserStore.getInstance());
+	    setConversationStore(ConversationStore.getInstance());
+	}
+	
+	/**
+	 * Sets the UserStore used by this servlet. This function provides a common setup method for use
+	 * by the test framework or the servlet's init() function.
+	 */
+	void setUserStore(UserStore userStore) {
+	  this.userStore = userStore;
+	}
+	
+	/**
+	 * Sets the ConversationStore used by this servlet. This function provides a common setup method
+	 * for use by the test framework or the servlet's init() function.
+	 */
+	void setConversationStore(ConversationStore conversationStore) {
+	  this.conversationStore = conversationStore;
+	}
+	
+	/**
+	 * This function fires when a user navigates to the activity feed page. It gets all of the
+	 * conversations and users from the model and forwards to activityfeed.jsp for rendering the list.
+	 * It also forwards the userStore so that activityfeed.jsp can find the owners of conversations.
+	 */
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response)
+		throws IOException, ServletException{
+	    List<Conversation> conversations = conversationStore.getAllConversations();
+	    List<User> users = userStore.getAllUsers();
+	    request.setAttribute("conversations", conversations);
```

Comment:
This is great if we want to show separate lists of conversations and users. It looks like in the view we are actually showing one list, one suggestion might be to add a data model that represents an "Activity" which we create from the users and the conversations, and then pass that as an attribute on the request rather than having separate attributes for users and conversations. I will explain how this can help us craft a better list in the view :)

Reply:
My original plan was to forward the three separate lists and then sort through them in the jsp file. I haven't written the code to sort through it yet so I just have the lists one after another. I considered having just an "Activity" type model, but then I figured I would need to create an interface and I wasn't sure if that would be changing too much of the structure.

Reply:
That could also work. Generally we try to limit the amount of logic in the view. If we have the logic in the servlet we can also write tests for it more easily. I don't think adding a new "Activity" model would be changing too much of the structure since you are adding an entirely new feature, so it is reasonable that with a new feature would come new models 😄

Reply:
Ok, I'll implement that instead! I think I'll just remove this functionality from my code so the activity feed is just a blank page for this pull request to make it simpler, and then I'll work on the new model for a later pull request.

Reply:
Not sure if this is helpful or not, but could we potentially save it as an ArrayList and sort it with a sorting array library function? Just wanted to throw an idea out there!

Reply:
Thanks for the tip! This is exactly what I was planning on doing, I'm just working on making an interface so that I can store instances of all 3 classes in the same List.